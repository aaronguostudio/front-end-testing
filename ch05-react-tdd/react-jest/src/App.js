import React from 'react';

function App() {
  return (
    <div className="app" title='aaron' data-test='container'>
      Hello World
    </div>
  );
}

export default App;
