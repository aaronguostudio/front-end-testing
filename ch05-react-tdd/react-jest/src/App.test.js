import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

Enzyme.configure({ adapter: new Adapter() })

it('renders without crashing', () => {
  // const div = document.createElement('div');
  // const container = div.getElementsByClassName('App')
  // expect(container.length).toBe(1)

  // shallow 只会渲染 App 这个组件，不会渲染内部的实现
  const wrapper = shallow(<App />)
  expect(wrapper).toMatchSnapshot()

  const container = expect(wrapper.find(`[data-test='container']`))

  // wrapper.debug() 可以把 dom 内容打印出来，帮助测试
  expect(wrapper.find('.app').length).toBe(1)
  expect(wrapper.find('.app').prop('title')).toBe('aaron')

  // 这样写，代码就不会和测试耦合了
  expect(wrapper.find(`[data-test='container']`).length).toBe(1)

  // jest-enzyme 提供的快捷语法
  container.toExist()
  container.toHaveProp('title', 'aaron')


});
