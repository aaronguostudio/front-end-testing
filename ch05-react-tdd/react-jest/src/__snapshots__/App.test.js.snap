// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="app"
  data-test="container"
  title="aaron"
>
  Hello World
</div>
`;
